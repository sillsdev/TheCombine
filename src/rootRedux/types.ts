import { type Action } from "redux";
import { type ThunkDispatch } from "redux-thunk";

import {
  type LoginState,
  defaultState as loginState,
} from "components/Login/Redux/LoginReduxTypes";
import {
  type CurrentProjectState,
  defaultState as currentProjectState,
} from "components/Project/ProjectReduxTypes";
import {
  type ExportProjectState,
  defaultState as exportProjectState,
} from "components/ProjectExport/Redux/ExportProjectReduxTypes";
import {
  type PronunciationsState,
  defaultState as pronunciationsState,
} from "components/Pronunciations/Redux/PronunciationsReduxTypes";
import {
  type TreeViewState,
  defaultState as treeViewState,
} from "components/TreeView/Redux/TreeViewReduxTypes";
import {
  type CharacterInventoryState,
  defaultState as characterInventoryState,
} from "goals/CharacterInventory/Redux/CharacterInventoryReduxTypes";
import {
  type MergeTreeState,
  defaultState as mergeDuplicateGoal,
} from "goals/MergeDuplicates/Redux/MergeDupsReduxTypes";
import {
  type GoalsState,
  defaultState as goalsState,
} from "goals/Redux/GoalReduxTypes";
import {
  type AnalyticsState,
  defaultState as analyticsState,
} from "types/Redux/analyticsReduxTypes";

/** Redux store structure */
export interface StoreState {
  //login
  loginState: LoginState;

  //project
  currentProjectState: CurrentProjectState;
  exportProjectState: ExportProjectState;

  //data entry and review entries
  treeViewState: TreeViewState;
  pronunciationsState: PronunciationsState;

  //goal timeline and current goal
  goalsState: GoalsState;

  //merge duplicates goal
  mergeDuplicateGoal: MergeTreeState;

  //character inventory goal
  characterInventoryState: CharacterInventoryState;

  //analytics state
  analyticsState: AnalyticsState;
}

/** Default values for the Redux store */
export const defaultState: StoreState = {
  loginState: { ...loginState },
  currentProjectState: { ...currentProjectState },
  exportProjectState: { ...exportProjectState },
  treeViewState: { ...treeViewState },
  pronunciationsState: { ...pronunciationsState },
  goalsState: { ...goalsState },
  mergeDuplicateGoal: { ...mergeDuplicateGoal },
  characterInventoryState: { ...characterInventoryState },
  analyticsState: { ...analyticsState },
};

export interface ActionWithPayload<T> extends Action {
  payload: T;
}

// https://redux.js.org/usage/usage-with-typescript#type-checking-redux-thunks
// suggests a custom general type for ThunkAction,
// so in like fashion, here's one for ThunkDispatch:
export type StoreStateDispatch = ThunkDispatch<StoreState, any, Action>;
