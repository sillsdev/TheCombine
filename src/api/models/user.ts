/* tslint:disable */
/* eslint-disable */
/**
 * BackendFramework
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

/**
 *
 * @export
 * @interface User
 */
export interface User {
  /**
   *
   * @type {string}
   * @memberof User
   */
  id: string;
  /**
   *
   * @type {string}
   * @memberof User
   */
  avatar: string;
  /**
   *
   * @type {boolean}
   * @memberof User
   */
  hasAvatar: boolean;
  /**
   *
   * @type {string}
   * @memberof User
   */
  name: string;
  /**
   *
   * @type {string}
   * @memberof User
   */
  email: string;
  /**
   *
   * @type {string}
   * @memberof User
   */
  phone: string;
  /**
   *
   * @type {string}
   * @memberof User
   */
  otherConnectionField?: string | null;
  /**
   *
   * @type {{ [key: string]: string; }}
   * @memberof User
   */
  workedProjects: { [key: string]: string };
  /**
   *
   * @type {{ [key: string]: string; }}
   * @memberof User
   */
  projectRoles: { [key: string]: string };
  /**
   *
   * @type {boolean}
   * @memberof User
   */
  agreement?: boolean;
  /**
   *
   * @type {string}
   * @memberof User
   */
  password: string;
  /**
   *
   * @type {string}
   * @memberof User
   */
  username: string;
  /**
   *
   * @type {string}
   * @memberof User
   */
  uiLang?: string | null;
  /**
   *
   * @type {string}
   * @memberof User
   */
  token: string;
  /**
   *
   * @type {boolean}
   * @memberof User
   */
  isAdmin: boolean;
}
