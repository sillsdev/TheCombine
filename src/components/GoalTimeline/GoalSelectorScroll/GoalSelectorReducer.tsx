import { StoreAction, StoreActions } from "../../../rootActions";
import { GoalSelectorState } from "../../../types/goals";
import {
  GoalScrollAction,
  MOUSE_ACTION,
  SELECT_ACTION,
} from "./GoalSelectorAction";

export const defaultState: GoalSelectorState = {
  selectedIndex: 0,
  allPossibleGoals: [],
  mouseX: 0,
  lastIndex: 0,
};

export const goalSelectReducer = (
  state: GoalSelectorState | undefined,
  action: StoreAction | GoalScrollAction
): GoalSelectorState => {
  if (!state) return defaultState;
  switch (action.type) {
    case SELECT_ACTION:
      return {
        ...state,
        selectedIndex: action.payload,
      };
    case MOUSE_ACTION:
      return {
        ...state,
        mouseX: action.payload,
      };
    case StoreActions.RESET:
      return defaultState;
    default:
      return state;
  }
};
