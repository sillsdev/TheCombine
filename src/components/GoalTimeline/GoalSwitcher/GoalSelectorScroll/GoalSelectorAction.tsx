export const SELECT_ACTION = "PRESS_BUTTON";
export type SELECT_ACTION = typeof SELECT_ACTION;

export const MOUSE_ACTION = "MOUSE_ACTION";
export type MOUSE_ACTION = typeof MOUSE_ACTION;

//action types
export interface ScrollSelectorAct {
  type: SELECT_ACTION;
  payload: number;
}
export interface MouseMoveAct {
  type: MOUSE_ACTION;
  payload: number;
}

export function scrollSelectorIndexAction(
  selectedIndex: number
): ScrollSelectorAct {
  return {
    type: SELECT_ACTION,
    payload: selectedIndex
  };
}

export function scrollSelectorMouseAction(mouseX: number): MouseMoveAct {
  return {
    type: MOUSE_ACTION,
    payload: mouseX
  };
}

export type ScrollAction = ScrollSelectorAct | MouseMoveAct;
