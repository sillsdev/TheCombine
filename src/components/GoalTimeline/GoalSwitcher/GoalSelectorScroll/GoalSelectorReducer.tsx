import {
  GoalScrollAction,
  SELECT_ACTION,
  MOUSE_ACTION
} from "./GoalSelectorAction";
import { GoalSelectorState } from "../../../../types/goals";

export const defaultState: GoalSelectorState = {
  selectedIndex: 0,
  goalOptions: [],
  mouseX: 0,
  lastIndex: 0
};

export const goalSelectReducer = (
  state: GoalSelectorState | undefined,
  action: GoalScrollAction
): GoalSelectorState => {
  if (!state) return defaultState;
  switch (action.type) {
    case SELECT_ACTION:
      return {
        ...state,
        selectedIndex: action.payload
      };

    case MOUSE_ACTION:
      return {
        ...state,
        mouseX: action.payload
      };

    default:
      return state;
  }
};
