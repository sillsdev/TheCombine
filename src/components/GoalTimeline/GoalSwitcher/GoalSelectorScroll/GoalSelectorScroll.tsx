import React, { ReactElement } from "react";

import Card from "@material-ui/core/Card";
import { Button } from "@material-ui/core";
import { Goal } from "../../../../types/goals";
import {
  LocalizeContextProps,
  withLocalize,
  Translate
} from "react-localize-redux";

export const WIDTH: number = 200;
export const VIEW_WIDTH: number = WIDTH * 3;
const MARGIN: number = 0;
const SELECT_COL = "cyan";

const SELECT = "selectstart";
const M_DUR = "mousemove";
const M_END = "mouseup";

// classNames
export const SCROLL_PANE: string = "scrollPane";
export const SCROLL_CONT: string = "scrollCont";
export const SCROLL_CARD: string = "scrollCard";
const PADDING: number = 10;

export interface GoalSelectorScrollProps {
  goalOptions: Goal[];
  selectedIndex: number;
  mouseX: number;
  lastIndex: number;
  swapSelectedIndex: (ndx: number) => void;
  swapMouseX: (iX: number) => void;
  handleChange: (value: string) => void;
}

export class GoalSelectorScroll extends React.Component<
  GoalSelectorScrollProps & LocalizeContextProps
> {
  // The set of styles used to make this UI work
  readonly style = {
    container: {
      display: "flex",
      flexWrap: "nowrap",
      overflow: "hidden"
    },
    pane: {
      padding: PADDING,
      userselect: "none",
      display: "flex",
      flexWrap: "nowrap",
      overflow: "hidden",
      width: VIEW_WIDTH
    },
    scroll: {
      flexWrap: "nowrap",
      display: "flex",
      width: this.getTickerWidth()
    },
    cardNormal: {
      width: WIDTH
    },
    cardSelect: {
      width: WIDTH,
      backgroundColor: SELECT_COL
    },
    cardWhitespace: {
      width: WIDTH,
      opacity: 0.01
    }
  };

  scrollRef: any;

  constructor(props: GoalSelectorScrollProps & LocalizeContextProps) {
    super(props);

    // Bind so as to be able to be used as event listeners/map function
    this.scrollDur = this.scrollDur.bind(this);
    this.scrollEnd = this.scrollEnd.bind(this);
    this.renderMapHelper = this.renderMapHelper.bind(this);
    this.scrollRef = React.createRef();
  }

  // Updates the position of the ticker upon mounting to override odd FireFox behavior
  componentDidMount() {
    try {
      this.getScroll().scrollLeft = 0;
    } catch (err) {
      // This block is only reached in a testing environment where the refs fail for reasons not currently understood. In
      // deployment, this block would never be reached (as under React's documentation, refs are guaranteed to work by the
      // time componentDidMount is called). Thus, this is a temporary solution to the tests not allowing refs to work
    }
  }

  // Creates and returns a single scrollCard based off of index
  renderMapHelper(goal: Goal, index: number, array: Goal[]): ReactElement {
    return this.scrollCard(index, this.style);
  }

  // Chooses an index based on the screen's current scroll
  selectNewIndex() {
    let w: number = this.getScroll().scrollLeft / WIDTH;
    if (w < 0) w += this.props.goalOptions.length;
    return Math.round(w);
  }

  // Pane scroll functions

  // Begin a scroll; creates additional listeners to allow it to be dragged outside of the narrow strip of scrolling pane.
  scrollStart(event: any) {
    window.addEventListener(SELECT, this.block);
    window.addEventListener(M_DUR, this.scrollDur);
    window.addEventListener(M_END, this.scrollEnd);
    this.props.swapMouseX(event.screenX);
  }

  // Track mouse during a scroll
  scrollDur(event: any) {
    this.scrollFreeform(this.props.mouseX - event.screenX);
    this.props.swapMouseX(event.screenX);
    this.props.swapSelectedIndex(this.selectNewIndex());
  }

  // End the scroll, remove listeners, lock into proper position
  scrollEnd(event: any) {
    window.removeEventListener(SELECT, this.block);
    window.removeEventListener(M_DUR, this.scrollDur);
    window.removeEventListener(M_END, this.scrollEnd);
    this.scrollLockNdx(this.selectNewIndex());
  }

  // Prevent selection of text while scrolling
  block(event: Event) {
    event.preventDefault();
  }

  // A lot of functions which handle the actual mechanics of scrolling

  // Scroll left one goal (w/ wrap)
  scrollLeft(): void {
    let newNdx = this.props.selectedIndex - 1;
    this.scrollLockNdx(newNdx < 0 ? this.props.goalOptions.length - 1 : newNdx);
  }

  // Scroll right one goal (w/ wrap)
  scrollRight(): void {
    let newNdx = (this.props.selectedIndex + 1) % this.props.goalOptions.length;
    this.scrollLockNdx(newNdx);
  }

  // Set this.props.selectedIndex and the current centered goal to a specified index
  scrollLockNdx(newIndex: number): void {
    this.props.swapSelectedIndex(newIndex);
    this.scrollToIndex(newIndex);
  }

  // Scroll to a specified new index
  scrollToIndex(newIndex: number): void {
    this.getScroll().scrollLeft = WIDTH * newIndex;
  }

  // Scroll the pane freely w/o locking on to goals
  scrollFreeform(amount: number): void {
    var el: HTMLElement = this.getScroll();
    var newScrollLeft = el.scrollLeft + amount;
    if (newScrollLeft < 0)
      newScrollLeft += this.getTickerWidth() - VIEW_WIDTH - PADDING - 1;
    else if (
      newScrollLeft >
      this.getTickerWidth() - VIEW_WIDTH - PADDING + MARGIN
    )
      newScrollLeft = 1;
    el.scrollLeft = newScrollLeft;
  }

  // Get the scrolling pane itself
  getScroll(): HTMLElement {
    return this.scrollRef.current;
  }

  // Get the width of the scrolling component
  getTickerWidth(): number {
    return WIDTH * (this.props.goalOptions.length + 2);
  }

  // End of scrolling mechanic functions

  // Create the cards used to display the goals
  // For later: add in a change-color effect on mouse hover for the currently selected card to help indicate you can click on it
  scrollCard(index: number, style: any): ReactElement {
    return (
      <Card
        className={SCROLL_CARD}
        key={index}
        style={
          this.props.selectedIndex == index
            ? style.cardSelect
            : style.cardNormal
        }
        elevation={this.props.selectedIndex == index ? 7 : 1}
        onDoubleClick={() => {
          if (this.props.selectedIndex == index)
            this.props.handleChange(this.props.goalOptions[index].name);
        }}
      >
        <Translate id={"goal.name." + this.props.goalOptions[index].name} />
      </Card>
    );
  }

  render() {
    return (
      <div
        style={this.style.container as any}
        // Starts the drag event, then leaves it up to listeners created in scrollStart to see it through
        onMouseDown={(event: any) => {
          this.scrollStart(event);
        }}
      >
        {/* Scroll left button */}
        <Button
          onClick={(e: any) => {
            e.preventDefault();
            this.scrollLeft();
          }}
        >
          {"<"}
        </Button>
        {/* Scroll pane */}
        <div
          ref={this.scrollRef}
          className={SCROLL_PANE}
          id={SCROLL_PANE}
          style={this.style.pane as any}
        >
          <div
            className={SCROLL_CONT}
            id={SCROLL_CONT}
            style={this.style.scroll as any}
          >
            {/* Empty cards create whitespace at ends to allow for proper scrolling */}
            <Card style={this.style.cardWhitespace} />
            {this.props.goalOptions.map(this.renderMapHelper)}
            <Card style={this.style.cardWhitespace} />
          </div>
        </div>
        {/* Scroll right button */}
        <Button
          onClick={(e: any) => {
            e.preventDefault();
            this.scrollRight();
          }}
        >
          {">"}
        </Button>
      </div>
    );
  }
}

export default withLocalize(GoalSelectorScroll);
