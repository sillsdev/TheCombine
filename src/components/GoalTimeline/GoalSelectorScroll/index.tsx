import { connect } from "react-redux";

import { StoreState } from "../../../types";
import { StoreStateDispatch } from "../../../types/actions";
import { GoalSelectorState } from "../../../types/goals";
import {
  scrollSelectorIndexAction,
  scrollSelectorMouseAction,
} from "./GoalSelectorAction";
import GoalSelectorScroll from "./GoalSelectorScroll";

export function mapStateToProps(state: StoreState): GoalSelectorState {
  return {
    allPossibleGoals: state.goalsState.allPossibleGoals,
    selectedIndex: state.goalSelectorState.selectedIndex,
    mouseX: state.goalSelectorState.mouseX,
    lastIndex: state.goalsState.allPossibleGoals.length - 1,
  };
}

export function mapDispatchToProps(dispatch: StoreStateDispatch) {
  return {
    swapSelectedIndex: (ndx: number) => {
      dispatch(scrollSelectorIndexAction(ndx));
    },
    swapMouseX: (iX: number) => {
      dispatch(scrollSelectorMouseAction(iX));
    },
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(GoalSelectorScroll);
