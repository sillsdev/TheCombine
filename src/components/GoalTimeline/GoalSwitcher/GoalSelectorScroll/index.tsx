import GoalSelectorScroll from "./GoalSelectorScroll";
import {
  GoalScrollAction,
  scrollSelectorIndexAction,
  scrollSelectorMouseAction,
} from "./GoalSelectorAction";

import { connect } from "react-redux";
import { StoreState } from "../../../../types";
import { GoalSelectorState } from "../../../../types/goals";
import { Dispatch } from "redux";

export function mapStateToProps(state: StoreState): GoalSelectorState {
  return {
    allPossibleGoals: state.goalsState.allPossibleGoals,
    selectedIndex: state.goalSelectorState.selectedIndex,
    mouseX: state.goalSelectorState.mouseX,
    lastIndex: state.goalsState.allPossibleGoals.length - 1,
  };
}

export function mapDispatchToProps(dispatch: Dispatch<GoalScrollAction>) {
  return {
    swapSelectedIndex: (ndx: number) => {
      dispatch(scrollSelectorIndexAction(ndx));
    },
    swapMouseX: (iX: number) => {
      dispatch(scrollSelectorMouseAction(iX));
    },
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(GoalSelectorScroll);
