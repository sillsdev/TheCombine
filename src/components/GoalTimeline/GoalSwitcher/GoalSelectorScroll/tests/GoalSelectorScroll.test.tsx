import React from "react";

import GoalSelectorScroll from "../";
import { Goal, GoalSelectorState } from "../../../../../types/goals";
import { BaseGoal } from "../../../../../types/baseGoal";
import { User } from "../../../../../types/user";
import { GoalSelectorScroll as GSScroll } from "../GoalSelectorScroll";
import { SELECT_ACTION } from "../GoalSelectorAction";

import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { act } from "react-dom/test-utils";
import renderer, {
  ReactTestInstance,
  ReactTestRenderer
} from "react-test-renderer";

const labels: string[] = ["handleDuplicates", "handleFlags", "grammarCheck"];

// Create the mock store
const gsState: GoalSelectorState = createTempState();
const storeState: any = {
  goalSelectorState: gsState,
  goalsState: { goalOptions: gsState.goalOptions }
};
const createMockStore = configureMockStore([thunk]);
const store = createMockStore(storeState);

// Mock the DOM
jest.autoMockOn();

// Variables used in testing: contain various parts of the UI
var scrollMaster: ReactTestRenderer;
var scrollHandle: ReactTestInstance;

beforeEach(() => {
  // Reset store actions
  store.clearActions();

  // Here, use the act block to be able to render our GoalState into the DOM
  act(() => {
    scrollMaster = renderer.create(
      <Provider store={store}>
        <GoalSelectorScroll handleChange={jest.fn()} />
      </Provider>
    );
    scrollHandle = scrollMaster.root.findByType(GSScroll);
  });
});

// Actual tests
describe("Testing the goal selector scroll ui", () => {
  it("Constructs correctly", () => {
    // Default snapshot test
    snapTest("default view");
  });

  it("Dispatches ndx to store on left click", () => {
    try {
      // This relies on the dispatch to the store getting called before the call to
      // getScroll(), which fails in the testing environment
      // Yes, it's jank, but we couldn't find a better way.
      scrollHandle.instance.scrollLeft();
    } catch (e) {
      expect(store.getActions()).toEqual([
        {
          type: SELECT_ACTION,
          payload: 2
        }
      ]);
    }
  });

  it("Dispatches ndx to store on right click", () => {
    try {
      // Same as scrollLeft
      scrollHandle.instance.scrollRight();
    } catch (e) {
      expect(store.getActions()).toEqual([
        {
          type: SELECT_ACTION,
          payload: 1
        }
      ]);
    }
  });
});

// Perform a snapshot test
function snapTest(name: string) {
  expect(scrollMaster.toJSON()).toMatchSnapshot();
}

// Create a usable temporary state, as opposed to a dummy state
function createTempState(): GoalSelectorState {
  let tempUser: User = {
    name: "Robbie",
    username: "NumbrOne",
    id: 1
  };
  let goals: Goal[] = [];

  for (let i: number = 0; i < labels.length; i++)
    goals[i] = createBGoal(i, labels[i], tempUser);

  return {
    selectedIndex: 0,
    goalOptions: [...goals],
    mouseX: 0,
    lastIndex: 3
  };
}

// Creates a BaseGoal with the specified attributes
function createBGoal(id: number, name: string, user: User): Goal {
  let goal: Goal = new BaseGoal();
  goal.id = id;
  goal.name = name;
  goal.user = user;
  return goal;
}
