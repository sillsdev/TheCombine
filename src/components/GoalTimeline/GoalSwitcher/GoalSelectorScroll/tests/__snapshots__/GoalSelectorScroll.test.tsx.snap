// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the goal selector scroll ui Constructs correctly 1`] = `
<div
  onMouseDown={[Function]}
  style={
    Object {
      "display": "flex",
      "flexWrap": "nowrap",
      "overflow": "hidden",
    }
  }
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      &lt;
    </span>
  </button>
  <div
    className="scrollPane"
    id="scrollPane"
    style={
      Object {
        "display": "flex",
        "flexWrap": "nowrap",
        "overflow": "hidden",
        "padding": 10,
        "userselect": "none",
        "width": 600,
      }
    }
  >
    <div
      className="scrollCont"
      id="scrollCont"
      style={
        Object {
          "display": "flex",
          "flexWrap": "nowrap",
          "width": 1000,
        }
      }
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
        style={
          Object {
            "opacity": 0.01,
            "width": 200,
          }
        }
      />
      <div
        className="MuiPaper-root MuiPaper-elevation7 MuiCard-root scrollCard MuiPaper-rounded"
        onDoubleClick={[Function]}
        style={
          Object {
            "backgroundColor": "cyan",
            "width": 200,
          }
        }
      >
        Missing translationId: goal.name.handleDuplicates for language: \${ languageCode }
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root scrollCard MuiPaper-rounded"
        onDoubleClick={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
      >
        Missing translationId: goal.name.handleFlags for language: \${ languageCode }
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root scrollCard MuiPaper-rounded"
        onDoubleClick={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
      >
        Missing translationId: goal.name.grammarCheck for language: \${ languageCode }
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
        style={
          Object {
            "opacity": 0.01,
            "width": 200,
          }
        }
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      &gt;
    </span>
  </button>
</div>
`;
